packages/dart/npt_flutter/lib/features/profile_list/bloc/profile_list_bloc.dart

```diff
@@ -36,8 +41,32 @@ class ProfileListBloc extends LoggingBloc<ProfileListEvent, ProfileListState> {
       emit(const ProfileListFailedLoad());
       return;
     }
-
     emit(ProfileListLoaded(profiles: profiles));
+
+    // Create ProfileBlocs
+    final profileBlocs = <ProfileBloc>[];
+    for (final uuid in profiles) {
+      final bloc = _profileCacheCubit.getProfileBloc(uuid);
+      profileBlocs.add(bloc);
+    }
+
+    // Wait for all ProfileBlocs to load their data
+    await Future.wait(
+      profileBlocs.map((bloc) async {
+        // Wait until the bloc is in a loaded state
+        await bloc.stream.firstWhere(
+          (state) => state is ProfileLoadedState || state is ProfileFailedLoad,
+          orElse: () => bloc.state,
+        );
+      }),
+    );
```

Comment:
What are you trying to achieve with this call?

Reply:
This block ensure all profile are loaded, so when sort is performed all the profiles are sorted correctly. Without that code block some profiled will be in `ProfileInitial` state making the profile property null. This means the sort function cannot access the profile name to sort the profile in ascending order by profile name. 

Triggering the sort event at the end of `onLoad` cause the profile to be sorted by profileName when the use access the connection screen without them having to click on the Profile Name header. Without it profile will be unsorted (sorted based on the order it was retrieved for the user's secondary)

Reply:
Why might one not want to block and wait for everything to load here?